### PR TITLE
Get token by id

### LINKED ISSUE #19: Implement API: GET /tokens/{id} (get token info by id)
https://github.com/Greenstand/treetracker-query-api/blob/5b796230237325ee526616d5d3d9d1281fbe8b92/docs/api/spec/query-api.yaml#L475-L485

To implement an endpoint:

* Please write an e2e test to cover it.
* Please follow the guide in the 'readme' to follow our architecture on the server-side.

---

Some hints:

- Please read our readme for more information/guide/tutorial.
- Here is [an engineering book](https://greenstand.gitbook.io/engineering/) in Greenstand.
- To know more about our organization, visit our [website](https://greenstand.org).
- If you want to join the slack community (some resources need the community member's permission), please leave your email address.

### PR DESCRIPTION
**Pull request by pierrelstan**
The following changes have been implemented:
- create e2e test & model test
- create Interface token 
- create TokensRepository 
- create tokens model
- create tokensRouter 

resolves  #19

### CODE REVIEW

**Comment by pierrelstan** [comment]
@dadiorchen, the e2e test passed successfully locally and npm run build too. 

But I noticed when I pushed my last changes to the branch,  the typescript compiles failed. 
Probably the issue comes from the configuration of typescript because where the error appears, the path is correct.

**Comment by dadiorchen** [comment]
@QuaidBartolomei I experienced something too about the path, my case is that locally, the path is wrong, BUT, it works, but when I upload to github, it failed, because of the wrong relative path.

File: server/routers/tokensRouter.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import express from 'express';
+import Joi from 'joi';
+import TokensModel from 'models/Tokens';
```
Comment (QuaidBartolomei) [inline server/routers/tokensRouter.ts:3]:
```suggestion
import TokensModel from 'models/tokens';
```

This caused a build error for me locally

File: server/models/Tokens.spec.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import TokensModel from './Tokens';
```
Comment (QuaidBartolomei) [inline server/models/Tokens.spec.ts:1]:
```suggestion
import TokensModel from './tokens';
```

Eslint detected this error

**Review by QuaidBartolomei — CHANGES_REQUESTED** [review]
You may need to run `npm install` to make sure your eslint and typescript packages are up to date.

**Review by QuaidBartolomei — CHANGES_REQUESTED** [review]
The filename of `models/token.ts` needs to be `models/Token.ts` to match the existing naming pattern in that directory. Can you revert those commits and change the filename instead?🙇  Sorry for being misleading before, I discovered the issue right away and did not initially think about the project structure. I should have edited those code comments once I realized.

**Comment by pierrelstan** [comment]
> The filename of `models/token.ts` needs to be `models/Token.ts` to match the existing naming pattern in that directory. Can you revert those commits and change the filename instead?🙇 Sorry for being misleading before, I discovered the issue right away and did not initially think about the project structure. I should have edited those code comments once I realized.

Don't worry about it, I will make the changes.

File: __tests__/e2e/tokens.spec.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+    expect(response.status).toBe(200);
+    expect(response.body).toMatchObject({
+      id: '24f4f5f7-c29e-4707-961a-3515be5a2f3e',
+      capture_id: 'f6c0e710-d80a-4e93-a9d2-d4edb52856af',
+      wallet_id: 'eecdf253-05b6-419a-8425-416a3e5fc9a0',
+      transfer_pending: false,
+      transfer_pending_id: null,
+      created_at: '2021-02-18T23:53:29.172Z',
+      updated_at: '2021-02-18T23:53:29.172Z',
```
Comment (QuaidBartolomei) [inline __tests__/e2e/tokens.spec.ts:18]:
```suggestion
   //   created_at: '2021-02-18T23:53:29.172Z',
   //  updated_at: '2021-02-18T23:53:29.172Z',
```

We will comment it out as a reminder to resolve this issue

**Review by QuaidBartolomei — CHANGES_REQUESTED** [review]
The test fails for me with this error:

```

  ● Tokens › /tokens/{tokenId}

    expect(received).toMatchObject(expected)

    - Expected  - 2
    + Received  + 2

      Object {
        "capture_id": "f6c0e710-d80a-4e93-a9d2-d4edb52856af",
        "claim": false,
    -   "created_at": "2021-02-18T23:53:29.172Z",
    +   "created_at": "2021-02-19T03:53:29.172Z",
        "id": "24f4f5f7-c29e-4707-961a-3515be5a2f3e",
        "transfer_pending": false,
        "transfer_pending_id": null,
    -   "updated_at": "2021-02-18T23:53:29.172Z",
    +   "updated_at": "2021-02-19T03:53:29.172Z",
        "wallet_id": "eecdf253-05b6-419a-8425-416a3e5fc9a0",
      }

       9 |
      10 |     expect(response.status).toBe(200);
    > 11 |     expect(response.body).toMatchObject({
         |                           ^
      12 |       id: '24f4f5f7-c29e-4707-961a-3515be5a2f3e',
      13 |       capture_id: 'f6c0e710-d80a-4e93-a9d2-d4edb52856af',
      14 |       wallet_id: 'eecdf253-05b6-419a-8425-416a3e5fc9a0',
```

It seems to be a timezone issue, I am 4 hours behind @pierrelstan and the test has a 4 hour difference. Removing those 2 lines passes the test.

**Review by QuaidBartolomei — APPROVED** [review]

**Comment by dadiorchen** [comment]
@pierrelstan thank you!

**Pull request merged**